2.adoc

```diff
@@ -72,5 +72,165 @@ Optional description
 . Minus A
 . Minus B
 
+=== Typora
+
+Dekstopový (Windows, macOS, Linux) komerčný markdown-only (GitHub flavored) editor s live preview dokumentu v rovnakom okne hneď počas písania. link:+https://typora.io+[https://typora.io]
+
+==== Porovnání rozdílů
+Jedná sa o desktopový editor, nie webovú komponentu. 
+Podporuje iba jazyk markdown (github-flavored). 
+Kvôli chýbajúcej command pallete vyžaduje používanie myši ak si užívateľ nepamätá klávesovú skratku. 
+Podporuje zmenu vzhľadu pomocou css šablón.
+
+
+==== Zhodnocení UI
+
+Defaultné rozhranie so zobrazenou outline dokumentu:
+
+image::https://i.imgur.com/n3vyN3i.png[]
+
+* Focus mode
+** Celý dokument vybledne, zvýraznený je len aktívny blok
+
+image::https://i.imgur.com/u6NGu0Z.png[]
+
+* typewriter mode
+** Zaručí aby aktívny riadok bol vždy v strede editoru. 
+
+* Source code mode
+  ** Namiesto priamo live renderingu je stále zobrazený zdrojový kód, ktorý je zvýraznený nie len farebne ale aj veľkosťou písma, odsadením (vlastne vyrenderovaný dokument kde je miesto textu stále markdown).  
+  ** Pri zapnutí tohto módu nefunguje drag n drop na pridanie obrázku do dokumentu
+
+image::https://i.imgur.com/NWJ2Mtz.png[]
+
+* Outline 
+  ** podporu zbalenia/rozbalenia je nutné zapnúť v nastaveniach
+  ** UX fail: linky len na texte, nie na celom riadku outline.
+  ** Pevná šírka 
+* Podpora viacerých tabov (UX fail: ale každý dokument ako aj nápovedu vyvolanú z menu otvorí v novom okne)
+* UX fail: defaultne neexistuje klávesová skratka na otvorenie nového tabu s dokumentom (štandardne cmd+t)
+* Chýbajú presety klavesových skratiek 
+* Nápoveda je formou markdown dokumentu 
+
+==== Pozorování UI v kontextu markup-editor
+* Chýba command paleta
+  ** Alternatívou ku klávesovým skratkám je klasické menu, kde je užívateľ nutený siahnúť po myši
+  ** klávesové skratky na styling dokumentu (nadpisy, odsadenie, tabulky, bloky kódu, citacie, zoznamy, foot notes, horizontal lines, bold, underline,..., vloženie obrázku) (neaktívne v source code mode)
+* Nepodporuje režim so zobrazeným zdrojákom a vyrenderovaným dokumentom vedľa seba
+  ** štandardne live rendering v tom istom okne (preview)
+  ** source code mod pridáva nadštandardný syntax highlighting zdrojaku (niečo ako vyrenderovaný dokument akurát je v ňom stále markdown)
+* Chýba statusbar
+  ** Chýba informácia o pozícií kurzora v dokumente
+  ** Auto-save na macOS v réžií systému, bez informácie o poslednom uložení
```

Comment:
Nemam mac, ale co znamena v rezii systemu? OS vidi ze ma nekdo otevreny descriptor na file a muze se obecne ptat programu pres nejake rozhrani na stav a ukaladat je sam?

Reply:
Netušim ako to vo vnútri funguje.
http://support.typora.io/Auto-Save/
https://developer.apple.com/library/content//documentation/DataManagement/Conceptual/DocBasedAppProgrammingGuideForOSX/StandardBehaviors/StandardBehaviors.html#//apple_ref/doc/uid/TP40011179-CH5-SW4

Reply:
Tak to vypada ze jo. To je docela cool :-)
Tim padem je to napsany dobre.